client/blocks/login/index.jsx

```diff
@@ -35,26 +36,30 @@ import Notice from 'components/notice';
 import PushNotificationApprovalPoller from './two-factor-authentication/push-notification-approval-poller';
 import userFactory from 'lib/user';
 import SocialConnectPrompt from './social-connect-prompt';
+import JetpackLogo from 'components/jetpack-logo';
 
 const user = userFactory();
 
 class Login extends Component {
 	static propTypes = {
+		isLinking: PropTypes.bool,
+		jetpack: PropTypes.bool.isRequired,
```

Comment:
Should the `jetpack` prop be required? Would it make any difference if we make it non-required, and remove it from `defaultProps`?

Reply:
I'm going to keep this as is. I believe that if we were stricter with prop could rely on expectations about their type, we could reduce complexity and eliminate some bugs. Knowing that this component must deal with `true | false` is preferable than needing to deal with `undefined | null | false | true`. `isRequired` in addition to a provided default also means that we'll have a warning if `undefined` or `null` were received, overriding `false` with other falsy values.

From a perfectly practical standpoint, it's nice for us devs to have have this property defined as a bool. If the default is removed, the property doesn't appear in React devtools when looking at the component. With the default, there's a simple toggle we can work with to see the behavior of the component:

![toggle](https://user-images.githubusercontent.com/841763/36020182-109c4a1c-0d82-11e8-8723-91f1b776c8c5.gif)

Reply:
Sounds good, thanks for the detailed explanation 👍 

I don't want to go into the strict vs non-strict discussion again 😅